root/app/Http/Controllers/HousingLoanChartController.php

```diff
@@ -0,0 +1,148 @@
+        $financialInstitutions = [
+            'financial_institution1',
+            'financial_institution2',
+            'financial_institution3',
+            'financial_institution4',
+        ];
+
+        $financialInstitutionCounts = [];
+        $financialInstitutionList = [];
+
+        //DBからデータ取得
+        foreach ($financialInstitutions as $financialInstitution) {
+            $financialInstitutionCount = HousingLoanChart::where($financialInstitution, 1)->count();
+            $financialInstitutionCounts[] = $financialInstitutionCount;
+        }
+
+        //集計結果の合計
+        $financialInstitutionTotal = array_sum($financialInstitutionCounts);
+
+        //chart.js用にデータの配列作成
+        foreach ($financialInstitutionCounts as $financialInstitutionCount) {
+            $financialInstitutionList[] = round(($financialInstitutionCount / $financialInstitutionTotal) * 100, 1);
+        }
+
+        return view('survey.index', [
+            'aggregateResultsOfUsage' => $aggregateResultsOfUsage,
+            'usageSituationTitleList' => $usageSituationTitleList,
+            'financialInstitutionList' => $financialInstitutionList,
+        ]);
+    }
+
+    public function questionPageIndex()
+    {
+        //質問TOPページ表示
+        return view('survey.question-page.index');
+    }
+
+    public function showPage1()
+    {
+        //page1表示
+        return view('survey.question-page.page1');
+    }
+    public function postPage1(FormPage1SessionRequest $request)
+    {
+
+        //page1のformに入力された値をセッションに保存
+        $request->session()->put('form.page1', $request->input('usage_situation'));
+        return redirect()->route('housing-loan.question-page.page2.showPage2');
+    }
+
+    public function showPage2(Request $request)
+    {
+        //page2表示
+        if ($request->session()->get('form.page1') == 3) {
+            return view('survey.question-page.page3');
+        } else {
+            return view('survey.question-page.page2');
+        }
+    }
+    public function postPage2(Request $request)
+    {
+        //page2のformに入力された値をセッションに保存
+        $page2FormData = $request->only(['financial_institution1', 'financial_institution2', 'financial_institution3', 'financial_institution4']);
+        $request->session()->put('form.page2', $page2FormData);
+        return redirect()->route('housing-loan.question-page.page3.showPage3');
+    }
+    public function showPage3(Request $request)
+    {
+        //page3表示
+        if (
+            isset($request->session()->get('form.page2')['financial_institution1']) || isset($request->session()->get('form.page2')['financial_institution2']) || isset($request->session()->get('form.page2')['financial_institution3']) || isset($request->session()->get('form.page2')['financial_institution4'])
+        ) {
+            return view('survey.question-page.page3');
+        } else {
+            return redirect()->route('housing-loan.question-page.page2.showPage2')->with('message', 'どれかお選びください。');
```

Comment:
[IMO]
ここでセッション情報がない場合、不正アクセスになると考えるので
例えば以下のようにエラーとしてレスポンスを返すと良さそうです。

```suggestion
            abort(409);
```

https://developer.mozilla.org/ja/docs/Web/HTTP/Status

https://readouble.com/laravel/10.x/ja/errors.html#http-exceptions

Reply:
アドバイスいただきありがとうございます。
上記のように修正しました。
[64dd49e](https://github.com/anna-nouchi-epkotsoftware/survey-site-laravel/pull/2/commits/64dd49e7a1a52eda142c08c97248e2e97d098b33)